contracts/UsingWitnet.sol

```diff
@@ -11,7 +11,7 @@ import "./WitnetRequestBoardProxy.sol";
  * @title The UsingWitnet contract
  * @notice Contract writers can inherit this contract in order to create Witnet data requests.
  */
-contract UsingWitnet {
+abstract contract UsingWitnet {
   using Witnet for Witnet.Result;
 
   WitnetRequestBoardProxy internal wrb;
```

Comment:
- For your consideration: if `wrb` was declared as 'WitnetRequestBoardInterface' instead of `WitnetRequestBoardProxy`, we would avoid client contract to forcely import (thus, having visibility) on the proxy implementation. **Pros**: this would partially mitigate audit's W02. **Cons**: client contract would have visibility on the `isUpgradable` method (declared in `WitnetRequestBoardInterface`), which is not actually supported in current implementation of the proxy (as it doesn't inherit from the interface). Any attempt of calling this method from the client contract (via `wrb`) would just revert.

Reply:
I think is a good idea to make wrb inmutable!

Reply:
Maybe we should use the Interface and make `wrb` immutable. =D